packages/cli/generation/remote-generation/remote-workspace-runner/src/SourceUploader.ts

```diff
@@ -0,0 +1,141 @@
+import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
+import { ApiDefinitionSource, SourceConfig } from "@fern-api/ir-sdk";
+import { loggingExeca } from "@fern-api/logging-execa";
+import { InteractiveTaskContext } from "@fern-api/task-context";
+import { FernWorkspace, IdentifiableSource } from "@fern-api/workspace-loader";
+import { FernRegistry as FdrAPI } from "@fern-fern/fdr-cjs-sdk";
+import { readFile, unlink } from "fs/promises";
+import tmp from "tmp-promise";
+
+const PROTOBUF_ZIP_FILENAME = "proto.zip";
+
+export type SourceType = "asyncapi" | "openapi" | "protobuf";
+
+export class SourceUploader {
+    private context: InteractiveTaskContext;
+    private sources: Record<string, IdentifiableSource>;
+    private sourceTypes: Set<SourceType>;
+
+    constructor(context: InteractiveTaskContext, workspace: FernWorkspace) {
+        this.context = context;
+        this.sources = Object.fromEntries(workspace.getSources().map((source) => [source.id, source]));
+        this.sourceTypes = new Set<SourceType>(Object.values(this.sources).map((source) => source.type));
+    }
+
+    public getSourceTypes(): Set<SourceType> {
+        return this.sourceTypes;
+    }
```

Comment:
nit: ive noticed that typescript classes often dont define a getter and just make the member public `public sourceTypes: Set<SourceType>` (i think its more idiomatic)